server.ts

```diff
@@ -108,10 +118,14 @@ export function AstroAuth(options = authConfig) {
  * @returns The current session, or `null` if there is no session.
  */
 export async function getSession(req: Request, options = authConfig): Promise<Session | null> {
+	if (isUserConfigLazy(options)) {
+		throw new Error(
+			'User Auth Configuration is Lazy. Fetch the session using getSessionByContext().'
+		)
+	}
```

Comment:
Since you're already checking for lazy config, we could have the functionality of `getSessionByContext` directly in here, right?
The `req` argument would need to be replaced with context (AstroGlobal, etc) which would be a breaking change, but I think that would be fine? Let me know what you think.